test/matrix_test.go

```diff
@@ -0,0 +1,561 @@
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  params:
+  - name: GOARCH
+    value: linux/amd64
+  - name: context
+    value: path/to/go117/context
+  - name: package
+    value: path/to/common/package/
+  - name: version
+    value: go1.17
+  serviceAccountName: default
+  timeout: 1h0m0s
+  taskRef:
+    name: mytask
+    kind: Task
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+`, "pr-matrix-include-0", namespace))
+
+	taskRun2 := parse.MustParseV1TaskRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  params:
+  - name: GOARCH
+    value: linux/ppc64le
+  - name: context
+    value: path/to/go117/context
+  - name: package
+    value: path/to/common/package/
+  - name: version
+    value: go1.17
+  serviceAccountName: default
+  timeout: 1h0m0s
+  taskRef:
+    name: mytask
+    kind: Task
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+`, "pr-matrix-include-1", namespace))
+
+	taskRun3 := parse.MustParseV1TaskRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  params:
+  - name: GOARCH
+    value: linux/s390x
+  - name: context
+    value: path/to/go117/context
+  - name: flags
+    value: -cover -v
+  - name: package
+    value: path/to/common/package/
+  - name: version
+    value: go1.17
+  serviceAccountName: default
+  timeout: 1h0m0s
+  taskRef:
+    name: mytask
+    kind: Task
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+`, "pr-matrix-include-2", namespace))
+
+	taskRun4 := parse.MustParseV1TaskRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  params:
+  - name: GOARCH
+    value: linux/amd64
+  - name: package
+    value: path/to/common/package/
+  - name: version
+    value: go1.18.1
+  serviceAccountName: default
+  timeout: 1h0m0s
+  taskRef:
+    name: mytask
+    kind: Task
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+`, "pr-matrix-include-4", namespace))
+
+	taskRun5 := parse.MustParseV1TaskRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  params:
+  - name: GOARCH
+    value: linux/ppc64le
+  - name: package
+    value: path/to/common/package/
+  - name: version
+    value: go1.18.1
+  serviceAccountName: default
+  timeout: 1h0m0s
+  taskRef:
+    name: mytask
+    kind: Task
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+`, "pr-matrix-include-4", namespace))
+
+	taskRun6 := parse.MustParseV1TaskRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  params:
+  - name: GOARCH
+    value: linux/s390x
+  - name: flags
+    value: -cover -v
+  - name: package
+    value: path/to/common/package/
+  - name: version
+    value: go1.18.1
+  serviceAccountName: default
+  timeout: 1h0m0s
+  taskRef:
+    name: mytask
+    kind: Task
+status:
+  conditions:
+  - status: "True"
+    type: Succeeded
+`, "pr-matrix-include-5", namespace))
+
+	taskRun7 := parse.MustParseV1TaskRun(t, fmt.Sprintf(`
```

Comment:
can we generate fewer fanned out tasks to reduce test latency, since each task results in a pod?

Reply:
Yep! Took out a few matrix params / taskRuns

Reply:
I still see 5 matrixed taskruns, is this intentional?

Reply:
Yes, from 7 --> 5. Can take out more if needed..